BUILD.gn

```diff
@@ -129,6 +129,11 @@ deno_cargo_info = exec_script("build_extra/rust/get_cargo_info.py",
                               [ rebase_path("Cargo.toml", root_build_dir) ],
                               "json")
 
+# Reads the typescript info from package.json
+deno_ts_info = exec_script("build_extra/rust/get_ts_info.py",
```

Comment:
Hrm. This works, but it's kinda ad hoc... We have a similar situation with `deno.platform` where we need to inject some information into the build. It's done using rollup here:
https://github.com/denoland/deno/blob/748b0f9c9d267dddf6f59c9d2ca8728bc76e630f/rollup.config.js#L92-L124
What do you think about doing something similar for the TS version?

Reply:
Yeah it should also work (this platform injection was also done by me quite a while ago). Probably we can centralize all the inject variables during build in a single file.

Reply:
> though it sort of would be great not to have to go to JavaScript at all

I don't think that's worth optimizing for.

> Another problem I realized with GN injection is that it would not work with `cargo build`, unless specially handled in `build.rs`...

Good point

It would be good to have access to the versions in the runtime environment - maybe that can be done as part of this work... `deno.version.v8` for example would be nice to have.

Reply:
How about using [rollup-plugin-replace](https://github.com/rollup/rollup-plugin-replace) for embedding the version number in script? The plugin can replace certain types of symbols in scripts with the suplied values.